src/ORM/EagerLoading/DataListEagerLoader.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace SilverStripe\ORM\EagerLoading;
+
+use SilverStripe\ORM\DataList;
+use SilverStripe\ORM\QueryCache\DataQueryStoreInterface;
+use SilverStripe\ORM\DataObject;
+use InvalidArgumentException;
+use SilverStripe\Core\Injector\Injector;
+use Psr\Container\NotFoundExceptionInterface;
+
+class DataListEagerLoader implements QueryEagerLoaderInterface
```

Comment:
This would implement DataQueryExecutorInterface instead. It would take a cached query executor as a property, and pass all calls to that. performEagerLoading would be called from within its implementations of execute, getFirstRow, getLastRow, based on the eager relations specified in the FutureQueryHints object passed as an argument to those methods, rather than $this->eagerLoad.